tests/test_protocols/test_generator/test_common.py

```diff
@@ -0,0 +1,371 @@
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the tests for generator/common.py module."""
+import logging
+import os
+import shutil
+import tempfile
+from pathlib import Path
+from unittest import TestCase
+
+from aea.protocols.generator.common import (
+    _camel_case_to_snake_case,
+    _create_protocol_file,
+    _get_sub_types_of_compositional_types,
+    _has_matched_brackets,
+    _includes_custom_type,
+    _match_brackets,
+    _python_pt_or_ct_type_to_proto_type,
+    _to_camel_case,
+    _union_sub_type_to_protobuf_variable_name,
+    load_protocol_specification,
+)
+
+from tests.test_protocols.test_generator.common import (
+    PATH_TO_T_PROTOCOL_SPECIFICATION,
+    T_PROTOCOL_NAME,
+)
+
+logger = logging.getLogger("aea")
+logging.basicConfig(level=logging.INFO)
+
+
+class TestCommon(TestCase):
+    """Test for generator/common.py."""
+
+    @classmethod
+    def setup_class(cls):
+        cls.cwd = os.getcwd()
+        cls.t = tempfile.mkdtemp()
+        os.chdir(cls.t)
+
+    def test_to_camel_case(self):
+        """Test the '_to_camel_case' method."""
+        input_text_1 = "this_is_a_snake_case_text"
+        expected_1 = "ThisIsASnakeCaseText"
+        output_1 = _to_camel_case(input_text_1)
+        assert output_1 == expected_1
+
+        input_text_2 = "This_is_a_Snake_Case_text"
+        expected_2 = "ThisIsASnakeCaseText"
+        output_2 = _to_camel_case(input_text_2)
+        assert output_2 == expected_2
+
+    def test_camel_case_to_snake_case(self):
+        """Test the '_camel_case_to_snake_case' method."""
+        input_text_1 = "ThisIsASnakeCaseText"
+        expected_1 = "this_is_a_snake_case_text"
+        output_1 = _camel_case_to_snake_case(input_text_1)
+        assert output_1 == expected_1
+
+    def test_match_brackets(self,):
+        """Positive test the '_match_brackets' method."""
+        text_1 = "[so[met[hi]]ng]"
+        assert _match_brackets(text_1, 0) == 14
+        assert _match_brackets(text_1, 3) == 11
+        assert _match_brackets(text_1, 7) == 10
+
+        text_2 = "[]]som[]et[hi[ng][sf]"
+        index_2 = 4
+        with self.assertRaises(SyntaxError) as cm:
+            _match_brackets(text_2, index_2)
+        self.assertEqual(
+            str(cm.exception),
+            "'index_of_open_bracket' in 'text' is not an open bracket '['. It is {}".format(
```

Comment:
Shouldn't the error be `"Index {} in 'text' is not an open bracket '['. It is {}".format(index, what_it_is)`?

Reply:
Done